lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -735,6 +820,22 @@ bool PrimUpdaterManager::mergeToUsd(const MFnDependencyNode& depNodeFn, const Uf
         return false;
     }
 
+    // Discard all pulled Maya nodes.
+    std::vector<MDagPath> toApplyOn = UsdMayaUtil::getDescendantsStartingWithChildren(mayaDagPath);
```

Comment:
Doesn't this mean that prim updaters that pull and generate both Dag and DG nodes will have to delete their DG nodes, but not their Dag nodes, because the Dag node delete is being done here?  That feels awkward, if that's the case.

Reply:
I think we have no choice. The Maya node have to be delete in the right order so that parent are not deleted before children, but individual updaters don't know what other updaters do. It seems safer to let the PrimUpdaterManager handle the deletions.

We might end-up having to manager the DG node deletions too. In fact, we will probably have to document very well what updater must do to be compatible with undo / redo.